manager/manager.go

```diff
@@ -204,14 +214,20 @@ func New(config *Config) (*Manager, error) {
 		grpc.Creds(config.SecurityConfig.ServerTLSCreds)}
 
 	m := &Manager{
-		config:      config,
-		listeners:   listeners,
-		caserver:    ca.NewServer(RaftNode.MemoryStore(), config.SecurityConfig),
-		Dispatcher:  dispatcher.New(RaftNode, dispatcherConfig),
-		server:      grpc.NewServer(opts...),
-		localserver: grpc.NewServer(opts...),
-		RaftNode:    RaftNode,
-		stopped:     make(chan struct{}),
+		config:                 config,
+		listeners:              listeners,
+		caserver:               ca.NewServer(RaftNode.MemoryStore(), config.SecurityConfig),
+		Dispatcher:             dispatcher.New(RaftNode, dispatcherConfig),
+		replicatedOrchestrator: orchestrator.NewReplicatedOrchestrator(RaftNode.MemoryStore()),
+		globalOrchestrator:     orchestrator.NewGlobalOrchestrator(RaftNode.MemoryStore()),
+		taskReaper:             orchestrator.NewTaskReaper(RaftNode.MemoryStore()),
+		scheduler:              scheduler.New(RaftNode.MemoryStore()),
+		keyManager:             keymanager.New(RaftNode.MemoryStore(), keymanager.DefaultConfig()),
```

Comment:
Many of these fields should only be initialized when the node is a leader.

Perhaps, it would be good to separate them into different structs based on leader and not leader.